File: Sort/ShellSort/ShellSort.java
package Sort.ShellSort;
public class ShellSort{
    void sort(int arr[]){
        int n = arr.length, x, i, j, temp;

        // x = big gap
        for( x = n/2; x > 0; x /= 2){
            // i = smaller gap
            for(i = x; i < n; i+=1){
                // temp array
                temp = arr[i];
                // shift values from smaller gap to correct location
                for(j = i; j >= x && arr[j-x] > temp; j -= x)
                arr[j] = arr[j - x];
                // move the temp array to correct location
                arr[j] = temp;

            }
        }
    }
    static void printArr(int arr[]){
        int n = arr.length, i;
        for(i = 0; i < n; ++i){
            System.out.println(arr[i] + " ");
            System.out.println();
        }
    }

    public static void main(String[]args){
        int arr[] = { 23, 45, 3, 5, 11 };
        ShellSort sort = new ShellSort();
        sort.sort(arr);
        printArr(arr);
    }
}

File: Sort/ShellSort/ShellSort.py
def ShellSort(arr,n):
    gap = n//2

    while gap > 0:
        for i in range(gap,n):
            temp = arr[i]
            j = i
            while j >= gap and arr[j - gap] > temp:
                arr[j] = arr[j-gap]
                j -= gap

            arr[j] = temp
        gap//=2

arr = [6, 1, 5, 15, 11, 9, 17, 16, 20, 14]
n = len(arr)
ShellSort(arr, n)
print(arr)
